### PR TITLE
Fix bug: Search terms containing trailing whitespace return no results.

### PR DESCRIPTION
**Pull request by leftclickben**
+ Remove extraneous whitespace from query string before splitting.
+ Also add missing tests for this module.

### CODE REVIEW

File: test/spec/util/convertBasicSearch.spec.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+(function () {
```
Comment (leftclickben) [inline test/spec/util/convertBasicSearch.spec.js]: For some reason this whole spec was missing, so here it is, with tests for extraneous whitespace.
